### ISSUE TITLE
Memory Leak / OOM with large map saves

### ISSUE THREAD

**Issue by restonic4** [sleepingdragoninn/surveyor#66]
The JVM constantly crashes because surveyor us using to much RAM, this happens to me on client side and on my dedicated server.

If I remove the surveyor folders on the server and client, it works again.

![image](https://github.com/user-attachments/assets/92055648-b287-4b8a-aabc-a2a8b121b464)

The exact same error everytime:

[14:03:44] [Render thread/INFO]: [Surveyor] Loading data for minecraft:overworld
[14:04:19] [Netty Client IO #3/ERROR]: Registry remapping failed!
java.util.concurrent.TimeoutException: null

....

[14:04:35] [Render thread/INFO]: [Surveyor] Loading data for minecraft:overworld
[14:04:36] [Netty Client IO #3/INFO]: Received common config 'fallingtrees:fallingtrees_common' from server
[14:05:08] [Render thread/INFO]: [FANCYMENU] ScreenCustomizationLayer registered: out_of_memory_screen
[14:05:09] [Render thread/ERROR]: Out of memory
java.lang.OutOfMemoryError: Java heap space

This happens frequently on really modded enviroments, but it also happens on vanilla clients (only the mod and antique atlas 4), its harder, but it happens.

If i create a new server with a void generator, just nothing, tick frozeen and nothing going on, if i explore the exact same amount of chunks with and without the mod those are the stats:

Without the mod: 3.6GB / 8GB
With the mod 7.8GB / 8GB

Client side log file while joining my dedicated server: 
[latest.log](https://github.com/user-attachments/files/18229763/latest.log)

**Comment by sleepingdragoninn** [comment]
Yep, this is #63 - without a proper streaming system, exploring chunks will linearly scale the amount of memory usage. You do have to try _really_ hard to generate untenable amounts of data (it's usually only possible in multiplayer or with a lot of time on your hands).
The size of your `world/data/surveyor` folder should be indicative: 400mb of data here (as retrieved from a server admin with 100+ players - I think it's about 8 million chunks? less if you have really big block palettes) equals roughly 24GB of memory usage in-game.
Because this is usually only possible on the server end or via sharing, you can set `terrain = DISABLED` in surveyor.toml ON THE SERVER ONLY as a workaround. Clients that already have too much save data will need to clear theirs from the data/surveyor folder.
